yeonjae/programmers-2/위장.js

```diff
@@ -0,0 +1,25 @@
+function solution(clothes) {
+  const countObj = {};
+  let answer = 1;
+
+  clothes.map((el) => {
+    if (countObj[el[1]]) {
+      return (countObj[el[1]] += 1);
+    }
+    return (countObj[el[1]] = 1);
+  });
+
+  const keys = Object.keys(countObj);
+
+  keys.forEach((el) => {
+    answer *= countObj[el] + 1;
+  });
+
+  return answer - 1;
+}
```

Comment:
```suggestion
function solution(clothes) {
  const countObj = {};
  let answer = 1;

  clothes.forEach(([_, clotheType]) => {
    countObj[clotheType] = (countObj[clotheType] || 0) + 1;
  });

  for (const type in countObj) {
    answer *= countObj[type] + 1;
  }
  
  return answer - 1;
}
```
- 풀이잘봤습니다 🙇🏻‍♂️
- map은 새로운 배열을 반환하고, 이를 활용하는 경우에 사용해주시고 지금과 같이 어떠한 배열에 할당한다거나 하지 않기때문에 forEach가 적절합니다!
- keys로 변환 후에 forEach보다 for...in문을 활용하면 좀 더 효율적으로 작성하실 수 있습니다.

Reply:
넵 차이점을 제대로 인지하지못하고있었던것같습니다. 코멘트 확인했습니다!